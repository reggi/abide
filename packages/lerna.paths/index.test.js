import mockFs from 'mock-fs'
import lernaPaths, {lernaCorePaths} from './index'

beforeEach(async () => {
  mockFs({
    '/working-case': {
      'packages': {
        'a': {'package.json': JSON.stringify({name: '@reggi/example-a'})},
        'b': {'package.json': JSON.stringify({name: '@reggi/example-b'})},
        'c': {'package.json': JSON.stringify({name: '@reggi/example-c'})}
      },
      'lerna.json': JSON.stringify({'packages': ['./packages/*']})
    },
    '/missing-packages-prop': {
      'packages': {
        'a': {'package.json': JSON.stringify({name: '@reggi/example-a'})},
        'b': {'package.json': JSON.stringify({name: '@reggi/example-b'})},
        'c': {'package.json': JSON.stringify({name: '@reggi/example-c'})}
      },
      'lerna.json': JSON.stringify({})
    }
  })
})

afterEach(async () => {
  mockFs.restore()
})

test('lernaCorePaths: working case', async () => {
  const result = await lernaCorePaths({workingDir: '/working-case'})
  expect(result).toMatchSnapshot()
})

test('lernaPaths: working case', async () => {
  const result = await lernaPaths({workingDir: '/working-case'})
  expect(result).toMatchSnapshot()
})

test('lernaPaths: missing packages prop', async () => {
  try {
    await lernaPaths({workingDir: '/missing-packages-prop'})
  } catch (e) {
    expect(e).toBeTruthy()
  }
  expect.assertions(1)
})
