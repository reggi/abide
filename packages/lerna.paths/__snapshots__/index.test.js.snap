// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lernaCorePaths: working case 1`] = `
Object {
  "binPath": "/working-case/node_modules/.bin/lerna",
  "configPath": "/working-case/lerna.json",
  "rootPackagePath": "/working-case/package.json",
  "rootPath": "/working-case",
}
`;

exports[`lernaPaths: working case 1`] = `
Object {
  "binPath": "/working-case/node_modules/.bin/lerna",
  "config": Object {
    "packages": Array [
      "./packages/*",
    ],
  },
  "configPath": "/working-case/lerna.json",
  "packagePaths": Array [
    "/working-case/packages/a",
    "/working-case/packages/b",
    "/working-case/packages/c",
  ],
  "rootPackagePath": "/working-case/package.json",
  "rootPath": "/working-case",
}
`;
